Arm backend: Add tests for int16 rsqrt on Ethos-U55/U85

Ninja91 opened the pull request:
Summary:
Fix Rsqrt op for int16
Relands D83802158

Differential Revision: D86402524




cc @freddan80 @per @zingo @oscarandersson8218 @digantdesai

SaoirseARM commented on backends/arm/test/ops/test_rsqrt.py:
```diff
@@ -104,3 +110,96 @@ def test_rsqrt_vgf_INT(test_tensor: torch.Tensor):
+
+
+@common.parametrize("test_tensor", Rsqrt.test_parameters)
+def test_rsqrt_int16_tosa_INT(test_tensor: torch.Tensor):
```
It looks like there are some failures on this particular testcase. For rand_4d_1 and rand _4d_2. It might be best to see these as expected failures like below. Thanks for adding this!

zingo commented:
Hi @Ninja91 this is the fails we see,

[pull / unittest-arm-backend-with-no-fvp (test_pytest_ops) / linux-job (pull_request)](https://github.com/pytorch/executorch/actions/runs/19161380882/job/54772221446?pr=15631)
FAILED backends/arm/test/ops/test_rsqrt.py::test_rsqrt_int16_tosa_INT[rand_4d_1]
FAILED backends/arm/test/ops/test_rsqrt.py::test_rsqrt_int16_tosa_INT[rand_4d_2]
FAILED backends/arm/test/ops/test_rsqrt.py::test_rsqrt_int16_tosa_INT[rand_3d]

 if you can break out the fails in it's own tests you can add a tag 
@pytest.mark.xfail( reason="text" )
for now, or just commen them out.

zingo commented on backends/arm/test/ops/test_rsqrt.py:
```diff
@@ -104,3 +110,96 @@ def test_rsqrt_vgf_INT(test_tensor: torch.Tensor):
+@pytest.mark.xfail(
+    reason="MLETORCH-707: AssertionError: Output 0 does not match reference output."
+)
+def test_rsqrt_int16_u55_INT16(test_tensor: torch.Tensor):
```
Test name should follow this naming convention   `test_<op>_16a8w_<target>`

Ninja91 committed:
Add tests for int16 rsqrt on Ethos-U55/U85 (#15631)

Summary:

Fix Rsqrt op for int16
Relands D83802158

Some TOSA INT16 tests are successful internally but failing in OSS. Marking them with xfail.
https://hud.pytorch.org/pr/pytorch/executorch/15631#54772221446

Differential Revision: D86402524

zingo commented:
Arm Backend Fail 
FAILED backends/arm/test/ops/test_sub.py::test_sub_tensor_tosa_INT_alpha[rand_4D_pi]
is unrelated and has a error rate on about 2-3% normaly we should retrigger to prove it but due to this is needed to be merged Ill just state it here this time.
Rest of the fails are also unrelated
